mx_unittest.py

```diff
@@ -36,14 +36,40 @@
 
 def _find_classes_with_annotations(p, pkgRoot, annotations, includeInnerClasses=False):
     """
-    Scan the sources of project 'p' for Java source files containing a line starting with 'annotation'
-    (ignoring preceding whitespace) and return the fully qualified class name for each Java
-    source file matched in a list.
+    Scan the sources of project 'p' for Java source files containing a line starting with
+    any element of 'annotations' (ignoring preceding whitespace) and return the list of fully
+    qualified class names for each Java source file matched.
     """
 
     matches = lambda line: len([a for a in annotations if line == a or line.startswith(a + '(')]) != 0
     return p.find_classes_with_matching_source_line(pkgRoot, matches, includeInnerClasses)
 
+def _find_classes_by_annotated_methods(annotations, suite):
+    """
+    Scan distributions from binary suite dependencies for classes contain at least one method
+    with an annotation from 'annotations' and return a dictionary from fully qualified class
+    names to the distribution containing the class.
+    """
+    binarySuiteDists = [d for d in mx.dependencies(opt_limit_to_suite=True) if d.isJARDistribution() and
+                        isinstance(d.suite, mx.BinarySuite) and (not suite or suite == d.suite)]
+    if len(binarySuiteDists) != 0:
+        # Ensure Java support class is built
+        mx.build(['--only', 'com.oracle.mxtool.junit'])
```

Comment:
I think we should use `--dependencies` instead of `--only` to avoid surprises regarding the dependencies of `com.oracle.mxtool.junit`.
For example, with `--only` the build could fail because some classes in `com.oracle.mxtool.junit` depend on the `JUNIT` library which is downloaded when using `--dependencies` but not necessarily when using `--only`.